integrationtest/mapper/orm/src/test/java/org/hibernate/search/integrationtest/mapper/orm/automaticindexing/AutomaticIndexingIdentiferRollbackIT.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.mapper.orm.automaticindexing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmUtils.withinTransaction;
+
+import java.util.concurrent.atomic.AtomicReference;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.util.impl.integrationtest.mapper.orm.OrmSetupHelper;
+import org.hibernate.search.util.impl.test.annotation.PortedFromSearch5;
+import org.hibernate.search.util.impl.test.annotation.TestForIssue;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test that Hibernate Search manages to delete entities even when {@code hibernate.use_identifier_rollback=true}.
+ */
+@TestForIssue(jiraKey = {"HSEARCH-650", "HSEARCH-3985"})
+@PortedFromSearch5(original = "org.hibernate.search.test.engine.UsingIdentifierRollbackTest")
+public class AutomaticIndexingIdentiferRollbackIT {
+
+	@Rule
+	public BackendMock backendMock = new BackendMock();
+
+	@Rule
+	public OrmSetupHelper ormSetupHelper = OrmSetupHelper.withBackendMock( backendMock );
+
+	private SessionFactory sessionFactory;
+
+	@Before
+	public void before() {
+		backendMock.expectAnySchema( EntityWithJpaIdAsDocumentId.NAME );
+		backendMock.expectAnySchema( EntityWithNonJpaIdAsDocumentId.NAME );
+		sessionFactory = ormSetupHelper
+				.start()
+				.withProperty( AvailableSettings.USE_IDENTIFIER_ROLLBACK, "true" )
+				.setup( EntityWithJpaIdAsDocumentId.class, EntityWithNonJpaIdAsDocumentId.class );
+		backendMock.verifyExpectationsMet();
+	}
+
+	@Test
+	public void jpaIdAsDocumentId() {
+		AtomicReference<Integer> entity1Id = new AtomicReference<>();
+
+		withinTransaction( sessionFactory, session -> {
+			EntityWithJpaIdAsDocumentId entity1 = new EntityWithJpaIdAsDocumentId();
+
+			session.persist( entity1 );
+
+			entity1Id.set( entity1.getId() );
+
+			backendMock.expectWorks( EntityWithJpaIdAsDocumentId.NAME )
+					.add( entity1.getId().toString(), b -> { } )
+					.processedThenExecuted();
+		} );
+		backendMock.verifyExpectationsMet();
+
+		AtomicReference<EntityWithJpaIdAsDocumentId> entity1FromTransaction = new AtomicReference<>();
```

Comment:
it seems that this reference is set in the lambda but never get after that

Reply:
Right, this was a remnant of a former attempt at testing this. It's not necessary, I removed it.